### ISSUE TITLE
distinguishing between sensitive and non-sensitive data

### ISSUE THREAD

**Issue by npdoty** [w3ctag/privacy-principles#159]
This has come up in a couple calls recently, but I think this principle in the personal data section may not work as a general principle:
> User agents should generally not attempt to distinguish between sensitive and nonsensitive personal data

I get the reasoning that follows, and there are related caveats in the following section on why it cannot easily or conclusively be determined what information might be sensitive to what person. But I wouldn't consider it contrary to the privacy principles of the web if my UA software tried to detect when I was entering a password or turning on my camera or pasting in what looked like a social security number or credit card number and took some additional action based on any of those rough heuristics about sensitivity. Browsers notably have done this with limiting some API access to secure contexts, for example, or I've seen features that prompt or notify the user about the privacy implications of entering an email address (or maybe offering a forwarding address) because of its capability as a persistent cross-origin identifier.

I would suggest that we remove this principle and just add to the text about caveats for determining sensitivity. Or that we explain in more detail what it means to "generally not attempt" this.

**Comment by pes10k** [comment]
Thanks for opening this @npdoty , i think it'd be best to discuss this on a call, but since i saw your comment about us having non-overlapping schedules for a while, I wanted to start conversation here

> But I wouldn't consider it contrary to the privacy principles of the web if my UA software tried to detect when I was entering a password or turning on my camera or pasting in what looked like a social security number or credit card number and took some additional action based on any of those rough heuristics about sensitivity

The intent of the current principals isn't that sites shouldn't collect or process data because of the blurry-and-or-non-boundary between sensitive and non-sensitive personal data.  I intended (though maybe poorly wrote) with the current text the following:

1. Sites should only collect personal data for purposes that help users achieve their goals
2. Its tricky-at-best (and possibly a fools-errand in practice) to draw a line between sensitive and non-sensitive data, and doubly so when the motivation is for sites to help themselves to the latter for non-user-serving reasons

In other words, i think the case you described would fall neatly in category 1.  The goal of the text isn't to say browsers shouldn't touch any data because all data is sensitive.

Or, more shortly, there are certainly lots of examples of data that are very sensitive (you mention many); but that doesn't imply the existence of non-sensitive data

**Comment by npdoty** [comment]
Ah, I thought the principle text was guidance or prohibition directed at user agents rather than sites. I didn't think it was a prohibition on browsers touching sensitive data, but I did think it was an explicit prohibition on ever providing different protections for different types of data or in different contexts.

Is the goal of category 2 maybe to note that we shouldn't assume that some category of data is completely innocuous and that no user would ever have interest in the privacy of it? That would make more sense to me: not that all user agent software should always consider all data types to be the same and never take different actions based on that (how I read the current text), but just recognizing that all data about a person could be data the user has an interest in keeping private.

**Comment by npdoty** [comment]
Revised data minimization section removes my concern, and a section on sensitive information remains. I think we can close this, but @pes10k may have other topics he wants to cover (which might be another issue).

**Comment by torgo** [comment]
We agreed to close today.
